[bug] tauri::Error is no longer Sync in 2.0.0-beta.0

### Describe the bug

As of 2.0.0-beta.0, the `tauri::Error` type no longer satisfies the `Sync` trait bound. It did previously in 2.0.0-alpha.21.

### Reproduction

```rust
use anyhow::Context;

#[cfg_attr(mobile, tauri::mobile_entry_point)]
pub fn run() -> anyhow::Result<()> {
	tauri::Builder::default()
		.run(tauri::generate_context!())
		.with_context("Failed to run Tauri app")
}
```

### Expected behavior

Likely for `tauri::Error` to still be `Sync` and for the code sample above to compile.

### Full `tauri info` output

```text
[✔] Environment
    - OS: Windows 10.0.22621 X64
    ✔ WebView2: 120.0.2210.144
    ✔ MSVC:
        - Visual Studio Build Tools 2022
        - Visual Studio Community 2022
    ✔ rustc: 1.75.0 (82e1608df 2023-12-21)
    ✔ cargo: 1.75.0 (1d8b05cdd 2023-11-20)
    ✔ rustup: 1.26.0 (5af9b9484 2023-04-05)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (default)
    - node: 20.10.0
    - npm: 10.2.3

[-] Packages
    - tauri [RUST]: 2.0.0-beta.1
    - tauri-build [RUST]: 2.0.0-beta.0
    - wry [RUST]: 0.35.2
    - tao [RUST]: 0.25.0
    - tauri-cli [RUST]: 2.0.0-alpha.21
    - @tauri-apps/api [NPM]: 2.0.0-beta.0
    - @tauri-apps/cli [NPM]: 2.0.0-beta.0

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../../ui/dist
    - devUrl: http://localhost:1420/
    - framework: Vue.js
    - bundler: Vite
```


### Stack trace

```text
error[E0599]: the method `with_context` exists for enum `Result<(), Error>`, but its trait bounds were not satisfied
   --> crates\tauri-app\src\lib.rs:7:4
    |
21  | /     tauri::Builder::default()
22  | |         .run(tauri::generate_context!())
23  | |         .with_context("Failed to run Tauri app")?;
    | |         -^^^^^^^^^^^^ method cannot be called on `Result<(), Error>` due to unsatisfied trait bounds
    | |_________|
    | 
    |
   ::: C:\Users\Schuyler\.cargo\registry\src\index.crates.io-6f17d22bba15001f\tauri-2.0.0-beta.1\src\error.rs:33:1
    |
33  |   pub enum Error {
    |   --------------
    |   |
    |   doesn't satisfy `tauri::Error: anyhow::context::ext::StdError`
    |   doesn't satisfy `tauri::Error: std::marker::Sync`
    |
   ::: C:\Users\Schuyler\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib/rustlib/src/rust\library\core\src\result.rs:502:1
    |
502 |   pub enum Result<T, E> {
    |   --------------------- doesn't satisfy `_: Context<(), Error>`
    |
    = note: the following trait bounds were not satisfied:
            `tauri::Error: anyhow::context::ext::StdError`
            which is required by `std::result::Result<(), tauri::Error>: anyhow::Context<(), tauri::Error>`
            `tauri::Error: std::marker::Sync`
            which is required by `std::result::Result<(), tauri::Error>: anyhow::Context<(), tauri::Error>`
```


### Additional context

This bound no longer being satisfied results in `tauri::Error` being incompatible with anyhow, since anyhow requires error types to be `std::error::Error + Send + Sync + 'static`.

--- a/.changes/tauri-error-sync.md
+++ b/.changes/tauri-error-sync.md
@@ -1,0 +1,5 @@
+---
+'tauri': 'patch:bug'
+---
+
+Fix regression in `tauri::Error` not being `Sync`.

--- a/.changes/tauri-scope-object-error-sync.md
+++ b/.changes/tauri-scope-object-error-sync.md
@@ -1,0 +1,5 @@
+---
+'tauri': 'patch:breaking'
+---
+
+Require `ScopeObject::Error` to be `Sync` as well.

--- a/core/tauri/src/error.rs
+++ b/core/tauri/src/error.rs
@@ -152,3 +152,12 @@ pub enum Error {
 
 /// `Result<T, ::tauri::Error>`
 pub type Result<T> = std::result::Result<T, Error>;
+
+#[cfg(test)]
+mod tests {
+  #[test]
+  fn error_is_send_sync() {
+    crate::test_utils::assert_send::<super::Error>();
+    crate::test_utils::assert_sync::<super::Error>();
+  }
+}

--- a/core/tauri/src/error.rs
+++ b/core/tauri/src/error.rs
@@ -147,7 +147,7 @@ pub enum Error {
   UnstableFeatureNotSupported,
   /// Failed to deserialize scope object.
   #[error("error deserializing scope: {0}")]
-  CannotDeserializeScope(Box<dyn std::error::Error + Send>),
+  CannotDeserializeScope(Box<dyn std::error::Error + Send + Sync>),
 }
 
 /// `Result<T, ::tauri::Error>`

--- a/core/tauri/src/ipc/authority.rs
+++ b/core/tauri/src/ipc/authority.rs
@@ -363,7 +363,7 @@ pub struct ScopeManager {
 /// though this is useful if you need to do some initialization logic on the type itself.
 pub trait ScopeObject: Sized + Send + Sync + Debug + 'static {
   /// The error type.
-  type Error: std::error::Error + Send;
+  type Error: std::error::Error + Send + Sync;
   /// Deserialize the raw scope value.
   fn deserialize<R: Runtime>(app: &AppHandle<R>, raw: Value) -> Result<Self, Self::Error>;
 }

